rclcpp/src/rclcpp/expand_topic_or_service_name.cpp

```diff
@@ -133,10 +134,14 @@ rclcpp::expand_topic_or_service_name(
           RCL_RET_ERROR, "failed to validate node name",
           rmw_get_error_state(), rmw_reset_error);
       }
-      throw rclcpp::exceptions::InvalidNodeNameError(
-              node_name.c_str(),
-              rmw_node_name_validation_result_string(validation_result),
-              invalid_index);
+
+      if (validation_result != RMW_NODE_NAME_VALID) {
+        throw rclcpp::exceptions::InvalidNodeNameError(
+                node_name.c_str(),
+                rmw_node_name_validation_result_string(validation_result),
+                invalid_index);
+      }
```

Comment:
Not sure if what scenario that can happen but should we do something in the else case ? (case where the rcl node name is valid but the rmw node name is not) (same everywhere below)

Reply:
this `if` is under umbrella of `RCL_RET_NODE_INVALID_NAME` and it's currently handling the cases of invalid rmw node name as you point :). or what you really mean the case that **rcl node name is invalid** but **rwm node name is valid** ?

Reply:
Indeed that's what I meant :smile:

Reply:
@mikaelarguedas now it's not sure when this will happen, I throw exceptions for the case of valid rmw with invalid rcl checking to allow the user to be aware of that for further investigation.